tensorflow_addons/seq2seq/sampler.py

```diff
@@ -400,8 +406,13 @@ class ScheduledOutputTrainingSampler(TrainingSampler):
     elsewhere.
     """
 
+    @typechecked
     def __init__(
-        self, sampling_probability, time_major=False, seed=None, next_inputs_fn=None
+        self,
+        sampling_probability: TensorLike,
+        time_major: bool = False,
+        seed: Optional[TensorLike] = None,
```

Comment:
```suggestion
        seed: Optional[int] = None,
```
Maybe too? I'm not sure.